### ISSUE TITLE
Two message steps with same name in one pipeline should not be allowed (and fails to finish)

### ISSUE THREAD

**Issue by e-gineer** [turbot/flowpipe#827]
I accidentally created a pipeline with two message steps with the same name:
```hcl
mod "local" {
  title = "flowpipe-mod-issues"
}

pipeline "two_message_steps_with_same_name" {

  param "key" {
    type = number
    default = 1
  }

  step "message" "test" {
    if = param.key % 2 == 0
    notifier = notifier.default
    text = "Even: ${param.key}"
  }

  step "message" "test" {
    if = param.key % 2 == 1
    notifier = notifier.default
    text = "Odd: ${param.key}"
  }

}
```

Current behavior is to execute the pipeline, but never exit / complete:
```
~/src/flowpipe-mod-issues $ flowpipe pipeline run sub_pipeline
[flowpipe] Execution ID: exec_co6q9q4o47mlidl94pug
[sub_pipeline] Starting pipeline
[sub_pipeline.test] Skipped
[sub_pipeline.test] Starting message: Odd: 1
[sub_pipeline.test] Arg text = Odd: 1
[sub_pipeline.test] Complete 2ms
^C
```

Expected behavior:
* Error at parse time - should not be able to define two steps with the same name.

**Comment by vhadianto** [comment]
Fixed in pipe-fittings: https://github.com/turbot/pipe-fittings/commit/83463a32a303bee62d86cd245d267f542a0497dc

**Comment by vhadianto** [comment]
This is released as part of v0.4.4
